Brain_I2C/Brain_I2C.ino

```diff
@@ -1,54 +1,146 @@
+/*
+ * Brain_I2C.ino - A c++ program that uses I2C to establish communication between 
+ * the sensors and valves inside to the rocket with the ground station. Able to send
+ * data to the ground station via RF. Can receive and process commands sent from 
+ * ground station. 
+ * Created by Vainavi Viswanath, Aug 21, 2020.
+ */
+
 #include <Wire.h>
+#include <SoftwareSerial.h>
+
+SoftwareSerial RFSerial(2,3);
 
 int board_address = 0;
 byte sensor_id = 0;
 uint8_t index = 0;
 
-void setup() {
-  Wire.begin();        // join i2c bus (address optional for master)
-  Serial.begin(9600);
-}
-
+/*
+ * Data structure to allow the conversion of bytes to floats and vice versa.
+ */
 union floatArrToBytes {
   char buffer[24];
   float sensorReadings[6];
 } farrbconvert;
 
+/*
+ * Data structure to store all information relevant to a specific sensor type.
+ */
 struct sensorInfo {
   String sensor_name;
   int board_address;
   byte sensor_id;
-  sensorInfo(String n, int b, byte s) : sensor_name(n), board_address(b), sensor_id(s) {}
+  int overall_id;
+  int clock_freq;
+  sensorInfo(String n, int b, byte s, int o, int f) : sensor_name(n), board_address(b), sensor_id(s), overall_id(o), clock_freq(f) {}
 };
 
-sensorInfo all_ids[2] = {sensorInfo("Pressure",8,0), sensorInfo("Temp",8,1)};
-sensorInfo sensor = sensorInfo("",0,0);
+/*
+ * Array of all sensors we would like to get data from.
+ */
+sensorInfo all_ids[8] = {
+  sensorInfo("Low Pressure",8,0,1,1), 
+  sensorInfo("High Pressure",8,1,2,2),
+  sensorInfo("Temperature",8,0,3,3),
+  sensorInfo("Load Cell",8,1,4,4),
+  sensorInfo("GPS",8,0,5,5),
+  sensorInfo("Barometer",8,1,6,6),
+  sensorInfo("Board Telemetry",8,0,7,7),
+  sensorInfo("Radio Telemetry",8,1,8,8)
+};
+
+sensorInfo sensor = sensorInfo("",0,0,0,0);
+
+/* 
+ *  Stores how often we should be requesting data from each sensor.
+ */
+int sensor_checks[sizeof(all_ids)/sizeof(sensorInfo)][2];
+
+void setup() {
+  Wire.begin();       
+  Serial.begin(9600);
+  RFSerial.begin(57600);
+  for (int i=0; i<sizeof(all_ids)/sizeof(sensorInfo); i++) {
+    sensor_checks[i][0] = all_ids[i].clock_freq;
+    sensor_checks[i][1] = 1;
+  }
+}
 
 void loop() {  
-  for (int j = 0; j < sizeof(all_ids)/sizeof(all_ids[0]); j++) {
+  for (int j = 0; j < sizeof(all_ids)/sizeof(sensorInfo); j++) {
+    if (sensor_checks[j][0] == sensor_checks[j][1]) {
+      sensor_checks[j][1] = 1;
+    } else {
+      sensor_checks[j][1] += 1;
+      delay(100);
+      continue;
```

Comment:
Say pressure should be read every 100 ms. , temp 200 ms, and GPS 3000 ms. Here all_ids[0] is the pressure, all_ids[1] is the temp, all_ids[2] is gps.
run through this code right here four times and tell me how often each of those sensors would read data.

You're very close, but not quite there. (the delay shouldn't be here, it should be outside of this for loop entirely)